### ISSUE TITLE
Listener "notebooks"

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#711]
Far out idea for sometime in the distant future.

The "notebook" concept from Mathematica and similar software might be a neat concept for a Factor listener.  Each statement can be executed changing the stack at that point.  If you go back and edit values, say changing a number from 1,000 to 2,000, the changes propagate as far as they can, assuming the stack effects allow them to.

Have builtin support for saving locals in the scope of the notebook, and this would be really handy.

**Comment by otoburb** [comment]
Once Project Jupyter (https://github.com/jupyter and http://jupyter.org/) gets off the ground it could be easier to have Factor as one of the languages supporting IPython-esque notebooks.

**Comment by catb0t** [comment]
@otoburb Jupyter is pretty mature now, but I don't think the Jupyter stack should be within this main repository, because from my understanding a new "Kernel" is made mostly of Python -- so it should be a separate repository under the Factor organisation, like **factor/factor-notebook** or something.

https://blog.jupyter.org/the-big-split-9d7b88a031a7
https://blog.jupyter.org/jupyter-ascending-1bf5b362d97e

**Comment by mrjbq7** [comment]
Agreed.  This would be a pretty "easy-ish" project, I would think since we have support for stuff like the emacs plugin and remote socket listeners.
